pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -236,7 +228,107 @@ func initFSS(ctx context.Context, k8sClient clientset.Interface, controllerClust
 		}
 	}
 
-	// Set up kubernetes resource listeners for k8s orchestrator
+	var doesSvFssCRExist bool
+	if controllerClusterFlavor == cnstypes.CnsClusterFlavorGuest {
+		var isFSSCREnabled bool
+		// Check if csi-sv-feature-states-replication FSS exists and is enabled
+		if val, ok := k8sOrchestratorInstance.internalFSS.featureStates[common.CSISVFeatureStateReplication]; ok {
+			isFSSCREnabled, err = strconv.ParseBool(val)
+			if err != nil {
+				log.Errorf("unable to convert %v to bool. csi-sv-feature-states-replication FSS disabled. Error: %v", val, err)
+				return err
+			}
+		} else {
+			log.Warn("csi-sv-feature-states-replication FSS not present")
+		}
+
+		// Initialize supervisor FSS map values in GC through CR if csi-sv-feature-states-replication FSS is enabled
+		if isFSSCREnabled {
+			cfg, err := common.GetConfig(ctx)
+			if err != nil {
+				log.Errorf("failed to read config. Error: %+v", err)
+				return err
+			}
+			// Get rest client config for supervisor
+			restClientConfig := k8s.GetRestClientConfigForSupervisor(ctx, cfg.GC.Endpoint, cfg.GC.Port)
+			// Get CNS operator client
+			cnsOperatorClient, err := k8s.NewClientForGroup(ctx, restClientConfig, cnsoperatorv1alpha1.GroupName)
+			if err != nil {
+				log.Errorf("failed to create CnsOperator client. Err: %+v", err)
+				return err
+			}
+
+			// Check if CnsCsiSvFeatureStates CR exists
+			svNamespace, err := cnsconfig.GetSupervisorNamespace(ctx)
+			if err != nil {
+				log.Errorf("failed to retrieve supervisor cluster namespace from config. Err: %+v", err)
+				return err
+			}
+			svFssCR := &featurestatesv1alpha1.CnsCsiSvFeatureStates{}
+			err = cnsOperatorClient.Get(ctx, client.ObjectKey{Name: featurestates.SVFeatureStateCRName,
+				Namespace: svNamespace}, svFssCR)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					// Log error and fallback to GCM replicated configmap approach
+					log.Errorf("CnsCsiSvFeatureStates CR not found in supervisor namespace %q. Err: %+v", svNamespace, err)
+				} else {
+					log.Errorf("failed to get CnsCsiSvFeatureStates CR from supervisor namespace. Err: %+v", svNamespace, err)
+					return err
+				}
+			} else {
+				doesSvFssCRExist = true
+				// Store SV FSS values in cache
+				for _, svFSS := range svFssCR.Spec.FeatureStates {
+					k8sOrchestratorInstance.supervisorFSS.featureStates[svFSS.Name] = strconv.FormatBool(svFSS.Enabled)
+				}
+				log.Infof("New supervisor feature states values stored successfully from %s CR object: %v",
+					featurestates.SVFeatureStateCRName, k8sOrchestratorInstance.supervisorFSS.featureStates)
+			}
+
+			// Set up namespaced listener for cnscsisvfeaturestate CR even if it doesn't exist
+			// This is required in cases where TKG is on a newer build and SV is being upgraded from an older version
+			// TODO: check if this will break backward compatibility as this CR is not yet registered in an old SV
+			err = k8sOrchestratorInstance.informerManager.AddFSSListener(ctx,
```

Comment:
Change the dynamic client to use supervisor client and test if it picks up new CR when SVC is upgraded

Reply:
Tested this and it works